### PR TITLE
[CSL-2337] no OutSpecs, Worker, ActionSpec

### PR DESCRIPTION
**Pull request by avieth**
## Description

These served only to confuse: the notion of `OutSpecs` only makes sense
in the context of the full diffusion layer. `OutSpecs` are unfortunately still
used, but they're tucked away in `Pos.Diffusion.Full`, where they are used to
describe the inter-node communication interface. You want to merge this
change because it means you don't have to deal with `OutSpecs`, which
were always `mempty` in practice.

With `OutSpecs` removed, the `*Spec` suffix of `ActionSpec` and `WorkerSpec`
is misleading, and their `Semigroup` instance became completely unnecessary
(it was used to combine the `OutSpecs`). These types were removed.

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: wallet-new/server/Main.hs [hunk 1]
```diff
@@ -82,8 +81,8 @@ actionWithWallet sscParams nodeParams ntpConfig wArgs@WalletBackendParams {..} =
         syncWallets
 
     runNodeWithInit ntpStatus init' nr =
-        let (ActionSpec f, outs) = runNode nr (plugins ntpStatus)
-         in (ActionSpec $ \s -> init' >> f s, outs)
+        let f = runNode nr (plugins ntpStatus)
+         in \s -> init' >> f s
```
Comment (parsonsmatt) [inline wallet-new/server/Main.hs]: I'd be tempted to simplify further to 

```haskell
runNodeWithInit ntpStatus init' nr s = do
  init'
  runNode nr (plugins ntpStatus) s
```

but this is already a big improvement :)

File: wallet-new/server/Main.hs [hunk 2]
```diff
@@ -123,16 +122,16 @@ actionWithNewWallet sscParams nodeParams params =
...
     runNodeWithInit
         :: PassiveWalletLayer Production
         -> NodeResources ext
-        -> (ActionSpec Kernel.Mode.WalletMode (), OutSpecs)
+        -> (Diffusion Kernel.Mode.WalletMode -> Kernel.Mode.WalletMode ())
     runNodeWithInit w nr =
-        let (ActionSpec f, outs) = runNode nr (plugins w)
-         in (ActionSpec $ \s -> f s, outs)
+        let f = runNode nr (plugins w)
+         in \s -> f s
```
Comment (parsonsmatt) [inline wallet-new/server/Main.hs]: this one can actually be simplified to `runNodeWithInit w nr = runNode nr (plugins w)`

**Comment by avieth** [comment]
I'd have added your suggestions to this patch but that would induce another 2 hour CI wait. Will make a followup later in the day

**Comment by parsonsmatt** [comment]
i know the feeling

**Pull request merged**